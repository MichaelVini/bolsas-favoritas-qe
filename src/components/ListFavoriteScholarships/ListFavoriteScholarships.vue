<template>
    <section class="card-favorite-scholarships">
        <AddCourseCard />
        <div v-for="(scholarship, index) in scholarshipsFilterBySemesters"  :key="index"  class="favorite-scholarships-list">
            <div class="scholarships-content">
                <div class="scholarship-content__logo">
                    <img :src="scholarship.university.logo_url" alt="" width="35%">
                </div>
                <span class="scholarships-content__span"><b>{{scholarship.university.name}}</b></span>
                <p class="scholarships-content__name-course">{{scholarship.course.name}}</p>
                <div class="scholarship-content__star-rating">
                    <img class="scholarship-content__star-rating__image" src="@/assets/star-rating.png" width="40%">
                    <p class="scholarships-content__text"><b>{{ scholarship.university.score }}</b></p>
                </div>
                <hr>
                <span class="scholarships-content__span"><b>{{scholarship.course.kind}} - {{ scholarship.course.shift}}</b></span>
                <p class="scholarships-content__text">Início das aulas em: {{ scholarship.start_date }}</p>
                <hr>
                <p class="scholarships-content__text">Mensalidade com o Quero Bolsa:</p>
                <p class="scholarships-content__text"><s>{{ scholarship.full_price }}</s></p>
                <h3 class="scholarships-content__h3" >R$ {{ scholarship.price_with_discount }}/mês</h3>
                <div class="favorite-scholarships-list__buttons">
                    <button class="btn btn--delete" @click="deleteScholarship(index)">Excluir</button>
                    <button class="btn btn--show-offers">Ver oferta</button>
                </div>
            </div>
        </div>
    </section>
</template>
<script>
import { mapMutations, mapState } from 'vuex';
import AddCourseCard from '../AddCourseCard/AddCourseCard.vue';
export default {
    components: { 
        AddCourseCard
    },
    methods: {
        ...mapMutations(["UPDATE_SHOWMODAL","UPDATE_FAVORITE_SCHOLARSHIPS"]),
        
        deleteScholarship(index){
            let arrayFavoriteScholarships = this.favoriteScholarships

            arrayFavoriteScholarships.splice(index, 1)
            this.UPDATE_FAVORITE_SCHOLARSHIPS(arrayFavoriteScholarships)
        },
        changeModal(){
            this.UPDATE_SHOWMODAL(true);
        }
    },
    computed: {
        ...mapState(
            ["semestersScholarships","favoriteScholarships"]),

        scholarshipsFilterBySemesters(){
            const filterBySecondSemester2019 = this.favoriteScholarships.filter((elem) => {
               return elem.enrollment_semester === "2019.2"  
            })

            const filterByFirstSemester2020 = this.favoriteScholarships.filter((elem) => {
               return elem.enrollment_semester === "2020.1"  
            })
            
            if(this.semestersScholarships[1] == true){
                return filterBySecondSemester2019
            } else if (this.semestersScholarships[2] == true) {
                return filterByFirstSemester2020
            } else {
                return this.favoriteScholarships
            }
        }
    }
}
</script>

<style src="./listFavoriteScholarships.css" scoped>