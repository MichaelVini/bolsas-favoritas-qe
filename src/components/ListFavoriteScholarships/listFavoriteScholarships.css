hr {
    border: 0;
    height: 2px;
    background-image: linear-gradient(to right, transparent, #CCC, transparent);
    margin: 15px 0; 
}

.card-favorite-scholarships {
    display: block;
    flex-wrap: wrap;
}
.favorite-scholarships-list {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 25px auto;
    box-shadow: rgba(99, 99, 99, 0.3) 0px 2px 8px 0px;
    max-width: 480px;
    min-height: 460px;
    padding: 30px 0;
    text-align: center;
}

.scholarships-content {
    width: 320px;
    margin: 0 auto;

}

.scholarships-content__name-course {
    font-weight: bold;
    color: var(--blue-secondary);
}

.scholarships-content__h3 {
   margin: 0;
   font-size: 1em;
   padding: 10px;
   color: var(--blue-secondary);
   font-weight: bold;
}

.scholarships-content__span {
    font-weight: 500;
    font-size: 1em;
}

.scholarships-content__text {
    margin: 0;
    color: black;
    font-weight: 200;
    margin-right: 5px;
    font-size: 1em;
}

.scholarship-content__logo {
    height: 110px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.scholarship-content__star-rating {
    display: flex;
    justify-content: center;
    align-items: center;
}

.scholarship-content__star-rating__image {
    margin: 2px 8px 5px 0px;
}

.favorite-scholarships-list__buttons {
    display: flex;
    margin-top: 30px;
}

.btn {
    border-radius: 4px;
    font-weight:bold;
    font-size: 0.95em;
}

.btn--delete {
    border: solid 2px var(--blue-primary);
    background-color: inherit;
    color: var(--blue-secondary);
    margin-right: 20px;
    transition: 0.4s;
    padding: 12px 32px;
    
}

.btn--delete:hover {
    background-color: var(--blue-primary);
    color: white;
    cursor: pointer;
}

.btn--show-offers {
    border: solid 2px rgb(226, 174, 4);
    background-color: rgb(243, 207, 2);
    transition: 0.3s;
    padding: 18px;
    width: 200px;
}

.btn--show-offers:hover {
    cursor: pointer;
    background-color: rgb(219, 187, 6);
}

@media (min-width: 1125px) {
    .card-favorite-scholarships{
        display: flex;
    }

    .scholarships-content__h3{
        color: rgb(2, 83, 2);
    }

    .scholarships-content__h3:nth-child(3){
        color: var(--blue-secondary);
    }

    .favorite-scholarships-list {
        margin: 0 auto;
        padding: 0;
    }

    .favorite-scholarships-list {
        display: flex;
        align-items: center;
        justify-content: center;
        margin: 25px auto;
        max-width: 480px;
        padding:0;
        text-align: center;
        margin: 0 30px 30px 0;
    }

    .scholarships-content__h3 {
        font-size: 0.9em;
     }
    
    .btn--delete {
        padding: 0;
        width: 90px;
        font-size: 0.9em;
        margin: 0 25px 0 40px;
    }
    
    .btn--show-offers {
        width: 135px;
        font-size: 0.9em;
        padding: 5px 0;
    }

    .favorite-scholarships-list__buttons {
        display: flex;
        margin-top: 10px;
    }
}