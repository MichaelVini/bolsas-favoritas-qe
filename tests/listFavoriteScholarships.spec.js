import { jest } from '@jest/globals'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import ListFavoriteScholarships from "./../src/components/ListFavoriteScholarships/ListFavoriteScholarships.vue"

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
    state:{
        semestersScholarships: [
            {
                "full_price": 2139.64,
                "price_with_discount": 706.08,
                "discount_percentage": 67.0,
                "start_date": "01/08/2019",
                "enrollment_semester": "2019.2",
                "enabled": true,
                "course": {
                  "name": "Engenharia",
                  "kind": "Presencial",
                  "level": "Bacharelado",
                  "shift": "Noite"
                },
                "university": {
                  "name": "UNIP",
                  "score": 4.5,
                  "logo_url": "https://www.tryimg.com/u/2019/04/16/unip.png"
                },
                "campus": {
                  "name": "Jardim das Indústrias",
                  "city": "São José dos Campos"
                }
              }
        ],
        favoriteScholarships: [
            {
                "full_price": 2139.64,
                "price_with_discount": 706.08,
                "discount_percentage": 67.0,
                "start_date": "01/08/2019",
                "enrollment_semester": "2019.2",
                "enabled": true,
                "course": {
                  "name": "Engenharia M",
                  "kind": "Presencial",
                  "level": "Bacharelado",
                  "shift": "Noite"
                },
                "university": {
                  "name": "UNIP",
                  "score": 4.5,
                  "logo_url": "https://www.tryimg.com/u/2019/04/16/unip.png"
                },
                "campus": {
                  "name": "Jardim das Indústrias",
                  "city": "São José dos Campos"
                }
            }
        ]
    },
    mutations: {
        UPDATE_SHOWMODAL: jest.fn()
    }
})

describe('ListFavoriteScholarships.vue', ()=> {
    it('should render default state', () => {
      const wrapper = shallowMount(ListFavoriteScholarships, {
        localVue,
        store
      })      
      
      expect(wrapper.html()).toMatchSnapshot()
    })
})
