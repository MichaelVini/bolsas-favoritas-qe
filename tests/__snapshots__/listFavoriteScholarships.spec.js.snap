// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFavoriteScholarships.vue should render default state 1`] = `
"<section class=\\"card-favorite-scholarships\\">
  <addcoursecard-stub></addcoursecard-stub>
  <div class=\\"favorite-scholarships-list\\">
    <div class=\\"scholarships-content\\">
      <div class=\\"scholarship-content__logo\\"><img src=\\"https://www.tryimg.com/u/2019/04/16/unip.png\\" alt=\\"\\" width=\\"35%\\"></div> <span class=\\"scholarships-content__span\\"><b>UNIP</b></span>
      <p class=\\"scholarships-content__name-course\\">Engenharia M</p>
      <div class=\\"scholarship-content__star-rating\\"><img src=\\"@/assets/star-rating.png\\" width=\\"40%\\" class=\\"scholarship-content__star-rating__image\\">
        <p class=\\"scholarships-content__text\\"><b>4.5</b></p>
      </div>
      <hr> <span class=\\"scholarships-content__span\\"><b>Presencial - Noite</b></span>
      <p class=\\"scholarships-content__text\\">Início das aulas em: 01/08/2019</p>
      <hr>
      <p class=\\"scholarships-content__text\\">Mensalidade com o Quero Bolsa:</p>
      <p class=\\"scholarships-content__text\\"><s>2139.64</s></p>
      <h3 class=\\"scholarships-content__h3\\">R$ 706.08/mês</h3>
      <div class=\\"favorite-scholarships-list__buttons\\"><button class=\\"btn btn--delete\\">Excluir</button> <button class=\\"btn btn--show-offers\\">Ver oferta</button></div>
    </div>
  </div>
</section>"
`;
